docs/blog-posts/symmetric-key-support.md

```diff
@@ -0,0 +1,220 @@
+# Adding Support for Symmetric Keys in Microsoft.Identity.Web
+
+## Overview
+This proposal outlines the addition of symmetric key support for signing credentials in Microsoft.Identity.Web, allowing keys to be loaded from Key Vault or Base64 encoded strings while maintaining clean abstractions.
+
+## Requirements
+1. Support symmetric keys from:
+   - Azure Key Vault secrets
+   - Base64 encoded strings
+2. Avoid circular dependencies with Microsoft.IdentityModel
+3. Follow existing patterns in the codebase
+4. Maintain backward compatibility
+
+## Developer Experience
+The implementation provides a straightforward and type-safe approach to working with symmetric keys while maintaining clean separation of concerns:
+
+### Key Management
+When working with symmetric keys, developers can utilize two primary sources:
+
+1. **Azure Key Vault Integration**
+   ```csharp
+   var credentials = new CredentialDescription
+   {
+       SourceType = CredentialSource.SymmetricKeyFromKeyVault,
+       KeyVaultUrl = "https://your-vault.vault.azure.net",
+       KeyVaultSecretName = "your-secret-name"
+   };
+   ```
+
+2. **Direct Base64 Encoded Keys**
+   ```csharp
+   var credentials = new CredentialDescription
+   {
+       SourceType = CredentialSource.SymmetricKeyBase64Encoded,
+       Base64EncodedValue = "your-base64-encoded-key"
+   };
+   ```
+
+### Implementation Details
+- The DefaultCredentialLoader automatically selects the appropriate loader based on the SourceType
+- Key material is loaded and converted to a SymmetricSecurityKey
+- The security key is stored in the CachedValue property of CredentialDescription
+- This design maintains independence from Microsoft.IdentityModel types in the abstractions layer
+- The implementation follows the same pattern as certificate handling for consistency
+
+## Design
+
+### 1. New CredentialSource Values(Abstractions Layer)
+```csharp
+public enum CredentialSource
+{
+    // Existing values
+    Certificate = 0,
+    KeyVault = 1,
+    Base64Encoded = 2,
+    Path = 3,
+    StoreWithThumbprint = 4,
+    StoreWithDistinguishedName = 5,
+
+    // New values
+    SymmetricKeyFromKeyVault = 6,
+    SymmetricKeyBase64Encoded = 7
+}
+```
+
+### 2. SymmetricKeyDescription Class(IdWeb Layer)
+Following the same pattern as CertificateDescription:
+
+```csharp
+public class SymmetricKeyDescription : CredentialDescription
```

Comment:
The `CredentialDescription` class already has the responsibility of managing different types of credentials - see the `CredentialType` property - it can handle Certificates, Signed Assertions (i.e. JWT tokens with a special audience to denote they are actually credentials), client secrets and decrypt keys. 

Have you considered just extending `CredentialDescription` instead of creating a new class? What are the pros / cons?